### PR TITLE
modify docker to start kafka and add kafka test

### PR DESCRIPTION
**Pull request by rasahner**
I modified the docker setup to start kafka in the same container as zookeeper.  (Putting them in separate containers adds a lot of complication and doesn't seem to add any value.)

I've added a test that uses the kafka firehose. It starts an indexing task using serverTime, sends some data using current timestamps, and verifies some queries.

I added a getKafkaHost() function to the configuration classes.

We have a test environment that uses multiple hosts, with zookeeper running on its own host.  When we ran integration tests there that use druid's service discovery code to find a realtime port, the port wasn't being found because service discovery was seeing localhost instead of the actual host as the zookeeper address.  To get around this, we bound CuratorConfig to a new class IntegrationTestingCuratorConfig that overrides getZkHosts() with a function that gets the zookeeper address from the config file.

### CODE REVIEW

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
+package io.druid.tests.indexer;
+
+/*
+This is a test for the kafka firehose.
+*/
+
+import com.google.api.client.repackaged.com.google.common.base.Throwables;
```
Comment (nishantmonu51) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: you can just import com.google.common.base.Throwables

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        Thread.sleep(DELAY_BETWEEN_EVENTS_SECS * 1000);
+      }
+      catch (InterruptedException ex) { /* nothing */ }
+      dtLast = dt;
+      dt = new DateTime(zone);
+    }
+
+    producer.close();
+
+    // wait for segments
```
Comment (nishantmonu51) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: we should also probably query the realtime task before segments are handed off in order to make sure realtime task are able to serve queries properly.

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java (hunk 1 above)
Comment (rasahner) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: done

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java (hunk 2 above)
Comment (rasahner) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: done

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
+package io.druid.tests.indexer;
```
Comment (himanshug) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: license header?

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+/*
+This is a test for the kafka firehose.
+*/
```
Comment (himanshug) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: i believe this is a class comment, conventionally we keep it close to class declaration statement, can you move it to line-40 before class declaration.

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+      int numPartitions = 1;
+      int replicationFactor = 1;
+      Properties topicConfig = new Properties();
+      AdminUtils.createTopic(zkClient, TOPIC_NAME, numPartitions, replicationFactor, topicConfig);
+    }
+    catch (TopicExistsException e) {
+      // it's ok if the topic already exists
+    }
+    catch (Exception e) {
+      LOG.error(e, "could not create kafka topic");
```
Comment (himanshug) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: shouldn't test fail here if topic creation failed for some reason?

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java (hunk 3 above)
Comment (rasahner) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: added

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java (hunk 4 above)
Comment (rasahner) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: moved

File: integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java (hunk 5 above)
Comment (rasahner) [inline integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaTest.java]: yes, it seems like is should.  I've changed it.

**Comment by himanshug** [comment]
LGTM

**Comment by rasahner** [comment]
taken care of by https://github.com/druid-io/druid/pull/1859

**Comment by drcrallen** [comment]
@rasahner can this be closed then?

**Pull request closed without merging**